interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: A reference to the object itself that the method is affecting 
 
-  Researched answer:
+  Researched answer: It refers to the object it belongs to  and has different values depending on where it is used. If it is in a method  it refers to the owner object, but alone it refers to the global object. 
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: It is a framework for creating a website that has added functionality and allows for easier scaling through the use of components for building the website
 
-  Researched answer:
+  Researched answer: Ract is a javascript library that helps build user interfaces using encapsulateed components that manage their own states and then composes them together to make a more complex UI.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The render method is required, otherwise it won't appear or be a part of the display
 
-  Researched answer:
+  Researched answer: render() is the only required method within a class component as it is what mounts and updates your component
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is the use of HTML tags within javascript functions, and is fully functional javascript but helps describe the display properties of the javascript elements
 
-  Researched answer:
+  Researched answer: JSX is a syntax extension to javascript and describes what the UI should look like. It is an extension of javascript that allows the suer to write HTML within JavaScript. HTML is supported by all browsers but JSX isn;t so it needs a compiler to make the browser understand it.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: yarn contains frameworks to help with creating and testing projects. yarn.lock, node modules
 
-  Researched answer:
+  Researched answer: yarn is a package manager that allows use of other developers solutions to problems. The package contains a file called package.json (called a manifest) that contains a description of the package.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.